### PR TITLE
Fixes unit tests in windows

### PR DESCRIPTION
**Pull request by logesr**
## Proposed changes

[//]: # (Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.)

## It fixes unit tests in windows. Resolves#[5950](https://github.com/webdriverio/webdriverio/issues/5950)

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [x ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/master/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have added proper type definitions for new commands (if appropriate)

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

**Review by mgrybyk — CHANGES_REQUESTED** [review]
Please enable windows tests https://github.com/webdriverio/webdriverio/blob/1bdc426ad9f0e380805de9489504ffd64a4344f7/.github/workflows/test.yml#L50

**Comment by mgrybyk** [comment]
Please fix failing tests

File: packages/wdio-appium-service/tests/launcher.test.js [hunk 1]
```diff
@@ -83,8 +84,12 @@ describe('Appium launcher', () => {
 
             expect(launcher.process).toBeInstanceOf(MockProcess)
             expect(launcher.logPath).toBe('./')
-            expect(launcher.command).toBe('path/to/my_custom_appium')
-            expect(launcher.appiumArgs).toMatchSnapshot()
+            if(isWindows){
```
Comment (logesr) [inline packages/wdio-appium-service/tests/launcher.test.js:87]: @mgrybyk  the failure is caused by the warnings thrown from obsolete snapshots. I have used a condition to validate snapshot only when the os is not windows. Also, we have separate snapshot test for windows. We don't have a way to remove these warnings. Only way is to use -u flag which updates snapshots everytime and passes automatically which is not a right way to do it. I am thinking to remove match snapshot from these tests and write separate snapshot tests for mac & linux like windows. Let me know your thoughts.

Comment (mgrybyk) [reply packages/wdio-appium-service/tests/launcher.test.js:87]: Separate snapshots should work, it's up to you if you have a better idea

**Comment by logesr** [comment]
@mgrybyk Thanks. I have fixed the unit tests. Smoke tests are failing because of package dependency. Any idea on this?

**Comment by mgrybyk** [comment]
I guess wrong path (linux style, I mean) is used to import the module. It should be adopted to work both in Linux and windows

**Comment by logesr** [comment]
Thanks @mgrybyk that really helped

File: packages/wdio-utils/src/utils.js [hunk 2]
```diff
@@ -159,7 +160,7 @@ export function safeRequire (name) {
          * also allows to link the package to a random place and have plugins
          * imported correctly (for dev purposes).
          */
-        const localNodeModules = `${process.cwd()}/node_modules`
+        const localNodeModules = path.join(`${process.cwd()}`, '/node_modules')
```
Comment (christian-bromann) [inline packages/wdio-utils/src/utils.js:163]:
```suggestion
        const localNodeModules = path.join(process.cwd(), 'node_modules')
```

File: packages/wdio-config/tests/configparser.test.js [hunk 3]
```diff
@@ -89,9 +100,16 @@ describe('ConfigParser', () => {
 
             const specs = configParser.getSpecs()
             expect(specs).toHaveLength(2)
-            let featureFileA = FIXTURES_CUCUMBER_FEATURE_A_LINE_2.split(':')[0]
+            let featureFileA = ''
+            let featureFileB = ''
+            if(isWindows){
```
Comment (christian-bromann) [inline packages/wdio-config/tests/configparser.test.js:105]:
```suggestion
            if (isWindows) {
```

File: packages/wdio-config/tests/configparser.test.js [hunk 4]
```diff
@@ -78,7 +84,12 @@ describe('ConfigParser', () => {
 
             const specs = configParser.getSpecs()
             expect(specs).toHaveLength(1)
-            let featureFileWithoutLine = FIXTURES_CUCUMBER_FEATURE_A_LINE_2_AND_12.split(':')[0]
+            let featureFileWithoutLine = ''
+            if(isWindows){
```
Comment (christian-bromann) [inline packages/wdio-config/tests/configparser.test.js:88]:
```suggestion
            if (isWindows) {
```

File: packages/wdio-config/tests/configparser.test.js [hunk 5]
```diff
@@ -67,7 +68,12 @@ describe('ConfigParser', () => {
 
             const specs = configParser.getSpecs()
             expect(specs).toHaveLength(1)
-            let featureFileWithoutLine = FIXTURES_CUCUMBER_FEATURE_A_LINE_2.split(':')[0]
+            let featureFileWithoutLine = ''
+            if(isWindows){
```
Comment (christian-bromann) [inline packages/wdio-config/tests/configparser.test.js:72]:
```suggestion
            if (isWindows) {
```

File: packages/wdio-appium-service/tests/launcher.test.js [hunk 6]
```diff
@@ -175,20 +185,53 @@ describe('Appium launcher', () => {
...
+        })
+
         test('should set correct config properties when empty', async () => {
             const launcher = new AppiumLauncher({}, [], {})
             await launcher.onPrepare()
 
             expect(launcher.logPath).toBe(undefined)
-            expect(launcher.command).toBe('node')
-            expect(launcher.appiumArgs).toMatchSnapshot()
+            if(isWindows){
```
Comment (christian-bromann) [inline packages/wdio-appium-service/tests/launcher.test.js:225]:
```suggestion
            if (isWindows) {
```

File: packages/wdio-appium-service/tests/launcher.test.js [hunk 7]
```diff
@@ -175,20 +185,53 @@ describe('Appium launcher', () => {
...
+                expect(launcher.command).toBe('cmd')
+            }else{
```
Comment (christian-bromann) [inline packages/wdio-appium-service/tests/launcher.test.js:227]:
```suggestion
            } else {
```

File: packages/wdio-appium-service/tests/launcher.test.js [hunk 8]
```diff
@@ -150,8 +157,11 @@ describe('Appium launcher', () => {
 
             expect(launcher.process).toBeInstanceOf(MockProcess)
             expect(launcher.logPath).toBe('./')
-            expect(launcher.command).toBe('path/to/my_custom_appium')
-            expect(launcher.appiumArgs).toMatchSnapshot()
+            if(isWindows){
```
Comment (christian-bromann) [inline packages/wdio-appium-service/tests/launcher.test.js:160]:
```suggestion
            if (isWindows) {
```

File: packages/wdio-appium-service/tests/launcher.test.js [hunk 9]
```diff
@@ -150,8 +157,11 @@ describe('Appium launcher', () => {
...
+                expect(launcher.command).toBe('cmd')
+            }else{
```
Comment (christian-bromann) [inline packages/wdio-appium-service/tests/launcher.test.js:162]:
```suggestion
            } else {
```

File: packages/wdio-appium-service/tests/launcher.test.js [hunk 10]
```diff
@@ -128,8 +132,11 @@ describe('Appium launcher', () => {
...
```
Comment (christian-bromann) [inline packages/wdio-appium-service/tests/launcher.test.js:137]:
```suggestion
            } else {
```

File: packages/wdio-appium-service/tests/launcher.test.js [hunk 11]
```diff
@@ -128,8 +132,11 @@ describe('Appium launcher', () => {
 
             expect(launcher.process).toBeInstanceOf(MockProcess)
             expect(launcher.logPath).toBe('./')
-            expect(launcher.command).toBe('path/to/my_custom_appium')
-            expect(launcher.appiumArgs).toMatchSnapshot()
+            if(isWindows){
```
Comment (christian-bromann) [inline packages/wdio-appium-service/tests/launcher.test.js:135]:
```suggestion
            if (isWindows) {
```

File: packages/wdio-appium-service/tests/launcher.test.js [hunk 12]
```diff
@@ -83,8 +84,11 @@ describe('Appium launcher', () => {
...
+                expect(launcher.command).toBe('cmd')
+            }else{
```
Comment (christian-bromann) [inline packages/wdio-appium-service/tests/launcher.test.js:89]:
```suggestion
            } else {
```

File: packages/wdio-appium-service/tests/launcher.test.js [hunk 13]
```diff
@@ -83,8 +84,11 @@ describe('Appium launcher', () => {
 
             expect(launcher.process).toBeInstanceOf(MockProcess)
             expect(launcher.logPath).toBe('./')
-            expect(launcher.command).toBe('path/to/my_custom_appium')
-            expect(launcher.appiumArgs).toMatchSnapshot()
+            if(isWindows){
```
Comment (christian-bromann) [inline packages/wdio-appium-service/tests/launcher.test.js:87]:
```suggestion
            if (isWindows) {
```

**Review by christian-bromann — CHANGES_REQUESTED** [review]
Some code formatting issues that we should add to eslint maybe

**Review by christian-bromann — APPROVED** [review]
Nice, thank you 👍

**Pull request merged**